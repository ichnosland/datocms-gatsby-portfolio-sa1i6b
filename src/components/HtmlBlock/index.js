/**
*
* HtmlBlock
*
*/

import React from 'react';

import PropTypes from 'prop-types';
import MathJax from 'react-mathjax-preview';
import styled from 'styled-components';

import Div from 'components/Div';
import { colore } from 'style/color';

export const HtmlBlockWrap = styled(Div)`
  width: 100%;
  color: ${(props) => props.color};
  background-color: ${(props) => props.bgcolor ? colore.ui.bgcolor : colore.ui.contrast};
`;

HtmlBlockWrap.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};


class HtmlBlock extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      text,
      hasLatex,
      color,
      bgcolor,
      align,
      padding,
      margin,
      className,
    } = this.props;
    return (
      <div>
        {hasLatex ?
          <HtmlBlockWrap
            padding={padding}
            margin={margin}
            align={align}
            color={color}
            bgcolor={bgcolor}
            className={className}
          >
            <MathJax math={text} />
          </HtmlBlockWrap>
          :
          <HtmlBlockWrap
            dangerouslySetInnerHTML={{
              __html: text,
            }}
            padding={padding}
            margin={margin}
            align={align}
            color={color}
            bgcolor={bgcolor}
            className={className}
          >
          </HtmlBlockWrap>
        }
      </div>
    );
  }
}

HtmlBlock.propTypes = {
  text: PropTypes.string,
  hasLatex: PropTypes.bool,
  color: PropTypes.string,
  align: PropTypes.string,
  bgcolor: PropTypes.string,
  padding: PropTypes.string,
  margin: PropTypes.string,
  className: PropTypes.string,
};

export default HtmlBlock;
