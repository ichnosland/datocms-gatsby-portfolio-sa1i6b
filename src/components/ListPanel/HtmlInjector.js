/**
*
* HtmlInjector
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import MathJax from 'react-mathjax-preview';
import CKEditor from 'react-ckeditor-component';

import media from 'style/mediainjector';
import Div from 'components/Div';
import { colore } from 'style/color';
import { HtmlStyles } from 'style/HtmlStylesVerifiche';

export const HtmlBlock = styled(Div)`
  color: ${(props) => props.color ? props.color : props.theme.brand};
  background-color: ${(props) => props.note ? colore.ui.note : colore.ui.contrast};
  border-top: 1px solid ${colore.ui.neutralBg};
  padding: 16px;
  ${media.print`
    border:none;
  `}
  ${HtmlStyles};
  overflow-y: ${(props) => props.overflowy};
  table {
    th {
      background: ${colore.ui.neutralBg};
    }
    td,
    th {
      padding: 0.62em;
      border: 1px solid ${colore.ui.neutralBg};
      vertical-align: middle;
    }
  }
  .numero-esercizio {
    float: right;
    background-color: ${colore.ui.darkBg};
    color: ${colore.ui.contrast};
    padding: 3px 6px;
    border-radius: 3px;
    text-align: right;
  }
`;

HtmlBlock.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const Solution = styled(HtmlBlock)`
  color: ${(props) => props.color ? props.color : '#0a6b06'};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : '#d0ffc8'};
`;

export const HtmlBlockFooter = styled(Div)`
  color: ${colore.ui.darkTxt};
  display: flex;
  justify-content: space-between;
  padding: 16px;
  border-top: 1px solid ${colore.ui.neutralBg};
  ${media.print`
    border:none;
  `}
`;

class HtmlInjector extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      text,
      hasLatex,
      solution,
      note,
      color,
      align,
      esercizi,
      data,
      editor = {},
    } = this.props;
    return (
      <Div>
        {text && editor.show &&
          <Div padding="16px" border="1px dashed rgb(251, 91, 111)" >
            <CKEditor
              activeClass="activeConsegnaEditor"
              content={editor.editorConsegnaHTML}
              events={{
                change: editor.onChangeConsegna,
              }}
              config={editor.config}
            />
          </Div>}
        {text && !editor.show && (!hasLatex ?
          <HtmlBlock
            dangerouslySetInnerHTML={{
              __html: text,
            }}
            align={align}
            note={note}
            color={color}
          >
          </HtmlBlock>
          :
          <HtmlBlock
            align={align}
            note={note}
            color={color}
            overflowy="auto"
          >
            <MathJax math={text} />
          </HtmlBlock>
        )}
        {solution && !editor.show && (!hasLatex ?
          <Solution
            dangerouslySetInnerHTML={{
              __html: solution,
            }}
          />
          :
          <Solution>
            <MathJax math={solution} />
          </Solution>
        )}
        {solution && editor.show &&
          <Solution>
            {<CKEditor
              activeClass="activeSoluzioneEditor"
              content={editor.editorSoluzioneHTML}
              events={{
                change: editor.onChangeSoluzione,
              }}
              config={editor.config}
            />}
          </Solution>}
        {(esercizi || data) &&
          <HtmlBlockFooter>
            {typeof esercizi !== 'undefined' &&
              <span>{`${esercizi} ${esercizi !== 1 ? 'ESERCIZI' : 'ESERCIZIO'}`}</span>
            }
            {data &&
              <span>{data}</span>
            }
          </HtmlBlockFooter>
        }
      </Div>
    );
  }
}

HtmlInjector.propTypes = {
  text: PropTypes.string,
  hasLatex: PropTypes.bool,
  solution: PropTypes.string,
  color: PropTypes.string,
  align: PropTypes.string,
  note: PropTypes.bool,
  esercizi: PropTypes.number,
  data: PropTypes.string,
  editor: PropTypes.shape({
    show: PropTypes.bool,
    editorConsegnaState: PropTypes.string,
    editorSoluzioneState: PropTypes.string,
    onChangeConsegna: PropTypes.func,
    onChangeSoluzione: PropTypes.func,
  }),
};

export default HtmlInjector;
