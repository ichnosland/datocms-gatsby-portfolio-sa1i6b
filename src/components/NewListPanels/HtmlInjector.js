/**
*
* HtmlInjector
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import MathJax from 'react-mathjax-preview';

import media from 'style/mediainjector';
import { HtmlBlockWrap } from '../HtmlBlock';
import { colore } from 'style/color';
import { ListItem, ItemTextStyle } from './index';
import { HtmlStyles } from 'style/HtmlStylesVerifiche';

export const HtmlBlock = styled(HtmlBlockWrap)`
  ${ItemTextStyle};
  background-color: ${(props) => props.note ? colore.ui.note : colore.ui.contrast};
  ${media.print`
    border:none;
  `}
  ${(props) => props.solution && css`
    color: #0a6b06;
    background-color: #d0ffc8;
  `};
  ${HtmlStyles};
`;

HtmlBlock.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};


class HtmlInjector extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      text,
      hasLatex,
      solution,
      note,
      color,
      align,
    } = this.props;
    return (
      <ListItem>
        {hasLatex ?
          <HtmlBlock
            align={align}
            note={note}
            color={color}
            solution={solution}
          >
            <MathJax math={text} />
          </HtmlBlock>
          :
          <HtmlBlock
            dangerouslySetInnerHTML={{
              __html: text,
            }}
            align={align}
            note={note}
            color={color}
            solution={solution}
          >
          </HtmlBlock>
        }
      </ListItem>
    );
  }
}

HtmlInjector.propTypes = {
  text: PropTypes.string,
  hasLatex: PropTypes.bool,
  solution: PropTypes.bool,
  color: PropTypes.string,
  align: PropTypes.string,
  note: PropTypes.bool,
};

export default HtmlInjector;
