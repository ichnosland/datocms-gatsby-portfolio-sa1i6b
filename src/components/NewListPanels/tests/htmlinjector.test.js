import React from 'react';
import { shallow } from 'enzyme';
import MathJax from 'react-mathjax-preview';

import { HtmlBlockWrap } from '../HtmlBlock';
import { ListItem } from '../index';
import HtmlInjector, { HtmlBlock } from '../HtmlInjector';

const text = 'del testo';

describe('<HtmlInjector />', () => {
  it('should render a ListItem', () => {
    const renderedComponent = shallow(<HtmlInjector />);
    expect(renderedComponent.find(ListItem).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should not render MathJax when hasLatex is false', () => {
    const hasLatex = false;
    const renderedComponent = shallow(<HtmlInjector hasLatex={hasLatex} text={text} />);
    expect(renderedComponent.find(MathJax).length).toEqual(0);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should not render MathJax when hasLatex is true', () => {
    const hasLatex = true;
    const renderedComponent = shallow(<HtmlInjector hasLatex={hasLatex} text={text} />);
    expect(renderedComponent.find(MathJax).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<HtmlBlock />', () => {
  it('should render a HtmlBlockWrap', () => {
    const renderedComponent = shallow(<HtmlBlock />);
    expect(renderedComponent.find(HtmlBlockWrap).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });
  it('should render its css w/ props align', () => {
    const align = 'left';
    const renderedComponent = shallow(<HtmlBlock align={align} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props color', () => {
    const color = 'red';
    const renderedComponent = shallow(<HtmlBlock color={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props note', () => {
    const renderedComponent = shallow(<HtmlBlock note />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props solution', () => {
    const renderedComponent = shallow(<HtmlBlock solution />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
