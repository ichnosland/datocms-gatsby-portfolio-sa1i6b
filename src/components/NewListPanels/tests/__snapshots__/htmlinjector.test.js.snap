// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HtmlBlock /> should render a HtmlBlockWrap 1`] = `
.c0 {
  padding: 22px 16px;
  background-color: rgb(255,255,255);
}

.c0 .segnaposto-admin {
  display: none;
}

.c0 .consegna__titolo {
  display: block;
  padding-bottom: 20px;
  font-weight: 600;
}

.c0 .consegna__contenuto {
  color: #000;
}

.c0 .opzioni {
  margin-top: 10px;
  padding: 10px;
  background-color: rgba(96,141,212,0.1);
  border: 1px solid rgba(35,56,89,.1);
  border-radius: 2px;
}

.c0 .opzioni__titolo {
  display: block;
  padding-bottom: 10px;
}

.c0 .opzioni__contenuto {
  color: #000;
  display: inline-block;
  font-style: italic;
  padding-left: 6px;
}

.c0 .opzioni__contenuto .radiocheck-wrap {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 0;
}

.c0 .opzioni .radio,
.c0 .opzioni .checkbox {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-right: 10px;
  padding-top: 3px;
}

.c0 .opzioni .radio:after,
.c0 .opzioni .checkbox:after {
  content: "";
  display: block;
  width: 14px;
  height: 14px;
  border: 1px solid #00BBEF;
  background: white;
}

.c0 .opzioni .checkbox:after {
  border-radius: 2px;
}

.c0 .opzioni .radio:after {
  border-radius: 100px;
}

.c0 .consegna__contenuto--esercizio {
  margin-top: 20px;
}

.c0 .completa {
  line-height: 2em;
}

.c0 .completa:after {
  content: "........................................";
  font-weight: 400;
}

.c0 .completa.full--width {
  display: block;
  height: 0;
  border-bottom: 1px dashed;
  margin: 2em 0;
}

.c0 .completa.full--width:after {
  content: "";
}

.c0 .completa.full--width:last-child {
  margin-bottom: 1em;
}

.c0 .mytestitaliano .contesto {
  border-bottom: 1px dotted;
}

.c0 .consegna__blocchi {
  margin-top: 10px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 4px;
  padding: 0 4px;
  border-radius: 2px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco span {
  padding: 4px 0;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza {
  border: 1px solid rgba(0,0,0,.4);
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza .segnaposto {
  border-bottom: 1px dashed rgba(0,0,0,.4);
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 p:last-child {
  margin-bottom: 0;
}

.c0 img {
  max-width: 100%;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

.c0 .editor {
  border: 1px dashed rgb(251,91,111);
}

.c0 .MathJax {
  font-size: 0.9em;
}

@media print {
  .c0 {
    border: none;
  }
}

@media print {
  .c0 .opzioni {
    background: none;
  }
}

<HtmlBlock__HtmlBlockWrap
  className="c0"
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
/>
`;

exports[`<HtmlBlock /> should render its css w/ props align 1`] = `
.c0 {
  padding: 22px 16px;
  background-color: rgb(255,255,255);
}

.c0 .segnaposto-admin {
  display: none;
}

.c0 .consegna__titolo {
  display: block;
  padding-bottom: 20px;
  font-weight: 600;
}

.c0 .consegna__contenuto {
  color: #000;
}

.c0 .opzioni {
  margin-top: 10px;
  padding: 10px;
  background-color: rgba(96,141,212,0.1);
  border: 1px solid rgba(35,56,89,.1);
  border-radius: 2px;
}

.c0 .opzioni__titolo {
  display: block;
  padding-bottom: 10px;
}

.c0 .opzioni__contenuto {
  color: #000;
  display: inline-block;
  font-style: italic;
  padding-left: 6px;
}

.c0 .opzioni__contenuto .radiocheck-wrap {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 0;
}

.c0 .opzioni .radio,
.c0 .opzioni .checkbox {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-right: 10px;
  padding-top: 3px;
}

.c0 .opzioni .radio:after,
.c0 .opzioni .checkbox:after {
  content: "";
  display: block;
  width: 14px;
  height: 14px;
  border: 1px solid #00BBEF;
  background: white;
}

.c0 .opzioni .checkbox:after {
  border-radius: 2px;
}

.c0 .opzioni .radio:after {
  border-radius: 100px;
}

.c0 .consegna__contenuto--esercizio {
  margin-top: 20px;
}

.c0 .completa {
  line-height: 2em;
}

.c0 .completa:after {
  content: "........................................";
  font-weight: 400;
}

.c0 .completa.full--width {
  display: block;
  height: 0;
  border-bottom: 1px dashed;
  margin: 2em 0;
}

.c0 .completa.full--width:after {
  content: "";
}

.c0 .completa.full--width:last-child {
  margin-bottom: 1em;
}

.c0 .mytestitaliano .contesto {
  border-bottom: 1px dotted;
}

.c0 .consegna__blocchi {
  margin-top: 10px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 4px;
  padding: 0 4px;
  border-radius: 2px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco span {
  padding: 4px 0;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza {
  border: 1px solid rgba(0,0,0,.4);
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza .segnaposto {
  border-bottom: 1px dashed rgba(0,0,0,.4);
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 p:last-child {
  margin-bottom: 0;
}

.c0 img {
  max-width: 100%;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

.c0 .editor {
  border: 1px dashed rgb(251,91,111);
}

.c0 .MathJax {
  font-size: 0.9em;
}

@media print {
  .c0 {
    border: none;
  }
}

@media print {
  .c0 .opzioni {
    background: none;
  }
}

<HtmlBlock__HtmlBlockWrap
  align="left"
  className="c0"
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
/>
`;

exports[`<HtmlBlock /> should render its css w/ props color 1`] = `
.c0 {
  color: red;
  padding: 22px 16px;
  background-color: rgb(255,255,255);
}

.c0 .segnaposto-admin {
  display: none;
}

.c0 .consegna__titolo {
  display: block;
  padding-bottom: 20px;
  font-weight: 600;
}

.c0 .consegna__contenuto {
  color: #000;
}

.c0 .opzioni {
  margin-top: 10px;
  padding: 10px;
  background-color: rgba(96,141,212,0.1);
  border: 1px solid rgba(35,56,89,.1);
  border-radius: 2px;
}

.c0 .opzioni__titolo {
  display: block;
  padding-bottom: 10px;
}

.c0 .opzioni__contenuto {
  color: #000;
  display: inline-block;
  font-style: italic;
  padding-left: 6px;
}

.c0 .opzioni__contenuto .radiocheck-wrap {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 0;
}

.c0 .opzioni .radio,
.c0 .opzioni .checkbox {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-right: 10px;
  padding-top: 3px;
}

.c0 .opzioni .radio:after,
.c0 .opzioni .checkbox:after {
  content: "";
  display: block;
  width: 14px;
  height: 14px;
  border: 1px solid #00BBEF;
  background: white;
}

.c0 .opzioni .checkbox:after {
  border-radius: 2px;
}

.c0 .opzioni .radio:after {
  border-radius: 100px;
}

.c0 .consegna__contenuto--esercizio {
  margin-top: 20px;
}

.c0 .completa {
  line-height: 2em;
}

.c0 .completa:after {
  content: "........................................";
  font-weight: 400;
}

.c0 .completa.full--width {
  display: block;
  height: 0;
  border-bottom: 1px dashed;
  margin: 2em 0;
}

.c0 .completa.full--width:after {
  content: "";
}

.c0 .completa.full--width:last-child {
  margin-bottom: 1em;
}

.c0 .mytestitaliano .contesto {
  border-bottom: 1px dotted;
}

.c0 .consegna__blocchi {
  margin-top: 10px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 4px;
  padding: 0 4px;
  border-radius: 2px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco span {
  padding: 4px 0;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza {
  border: 1px solid rgba(0,0,0,.4);
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza .segnaposto {
  border-bottom: 1px dashed rgba(0,0,0,.4);
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 p:last-child {
  margin-bottom: 0;
}

.c0 img {
  max-width: 100%;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

.c0 .editor {
  border: 1px dashed rgb(251,91,111);
}

.c0 .MathJax {
  font-size: 0.9em;
}

@media print {
  .c0 {
    border: none;
  }
}

@media print {
  .c0 .opzioni {
    background: none;
  }
}

<HtmlBlock__HtmlBlockWrap
  className="c0"
  color="red"
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
/>
`;

exports[`<HtmlBlock /> should render its css w/ props note 1`] = `
.c0 {
  padding: 22px 16px;
  background-color: #FFFDDE;
}

.c0 .segnaposto-admin {
  display: none;
}

.c0 .consegna__titolo {
  display: block;
  padding-bottom: 20px;
  font-weight: 600;
}

.c0 .consegna__contenuto {
  color: #000;
}

.c0 .opzioni {
  margin-top: 10px;
  padding: 10px;
  background-color: rgba(96,141,212,0.1);
  border: 1px solid rgba(35,56,89,.1);
  border-radius: 2px;
}

.c0 .opzioni__titolo {
  display: block;
  padding-bottom: 10px;
}

.c0 .opzioni__contenuto {
  color: #000;
  display: inline-block;
  font-style: italic;
  padding-left: 6px;
}

.c0 .opzioni__contenuto .radiocheck-wrap {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 0;
}

.c0 .opzioni .radio,
.c0 .opzioni .checkbox {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-right: 10px;
  padding-top: 3px;
}

.c0 .opzioni .radio:after,
.c0 .opzioni .checkbox:after {
  content: "";
  display: block;
  width: 14px;
  height: 14px;
  border: 1px solid #00BBEF;
  background: white;
}

.c0 .opzioni .checkbox:after {
  border-radius: 2px;
}

.c0 .opzioni .radio:after {
  border-radius: 100px;
}

.c0 .consegna__contenuto--esercizio {
  margin-top: 20px;
}

.c0 .completa {
  line-height: 2em;
}

.c0 .completa:after {
  content: "........................................";
  font-weight: 400;
}

.c0 .completa.full--width {
  display: block;
  height: 0;
  border-bottom: 1px dashed;
  margin: 2em 0;
}

.c0 .completa.full--width:after {
  content: "";
}

.c0 .completa.full--width:last-child {
  margin-bottom: 1em;
}

.c0 .mytestitaliano .contesto {
  border-bottom: 1px dotted;
}

.c0 .consegna__blocchi {
  margin-top: 10px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 4px;
  padding: 0 4px;
  border-radius: 2px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco span {
  padding: 4px 0;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza {
  border: 1px solid rgba(0,0,0,.4);
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza .segnaposto {
  border-bottom: 1px dashed rgba(0,0,0,.4);
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 p:last-child {
  margin-bottom: 0;
}

.c0 img {
  max-width: 100%;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

.c0 .editor {
  border: 1px dashed rgb(251,91,111);
}

.c0 .MathJax {
  font-size: 0.9em;
}

@media print {
  .c0 {
    border: none;
  }
}

@media print {
  .c0 .opzioni {
    background: none;
  }
}

<HtmlBlock__HtmlBlockWrap
  className="c0"
  note={true}
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
/>
`;

exports[`<HtmlBlock /> should render its css w/ props solution 1`] = `
.c0 {
  padding: 22px 16px;
  background-color: rgb(255,255,255);
  color: #0a6b06;
  background-color: #d0ffc8;
}

.c0 .segnaposto-admin {
  display: none;
}

.c0 .consegna__titolo {
  display: block;
  padding-bottom: 20px;
  font-weight: 600;
}

.c0 .consegna__contenuto {
  color: #000;
}

.c0 .opzioni {
  margin-top: 10px;
  padding: 10px;
  background-color: rgba(96,141,212,0.1);
  border: 1px solid rgba(35,56,89,.1);
  border-radius: 2px;
}

.c0 .opzioni__titolo {
  display: block;
  padding-bottom: 10px;
}

.c0 .opzioni__contenuto {
  color: #000;
  display: inline-block;
  font-style: italic;
  padding-left: 6px;
}

.c0 .opzioni__contenuto .radiocheck-wrap {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 0;
}

.c0 .opzioni .radio,
.c0 .opzioni .checkbox {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-right: 10px;
  padding-top: 3px;
}

.c0 .opzioni .radio:after,
.c0 .opzioni .checkbox:after {
  content: "";
  display: block;
  width: 14px;
  height: 14px;
  border: 1px solid #00BBEF;
  background: white;
}

.c0 .opzioni .checkbox:after {
  border-radius: 2px;
}

.c0 .opzioni .radio:after {
  border-radius: 100px;
}

.c0 .consegna__contenuto--esercizio {
  margin-top: 20px;
}

.c0 .completa {
  line-height: 2em;
}

.c0 .completa:after {
  content: "........................................";
  font-weight: 400;
}

.c0 .completa.full--width {
  display: block;
  height: 0;
  border-bottom: 1px dashed;
  margin: 2em 0;
}

.c0 .completa.full--width:after {
  content: "";
}

.c0 .completa.full--width:last-child {
  margin-bottom: 1em;
}

.c0 .mytestitaliano .contesto {
  border-bottom: 1px dotted;
}

.c0 .consegna__blocchi {
  margin-top: 10px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 4px;
  padding: 0 4px;
  border-radius: 2px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco span {
  padding: 4px 0;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza {
  border: 1px solid rgba(0,0,0,.4);
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza .segnaposto {
  border-bottom: 1px dashed rgba(0,0,0,.4);
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 p:last-child {
  margin-bottom: 0;
}

.c0 img {
  max-width: 100%;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

.c0 .editor {
  border: 1px dashed rgb(251,91,111);
}

.c0 .MathJax {
  font-size: 0.9em;
}

@media print {
  .c0 {
    border: none;
  }
}

@media print {
  .c0 .opzioni {
    background: none;
  }
}

<HtmlBlock__HtmlBlockWrap
  className="c0"
  solution={true}
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
/>
`;

exports[`<HtmlInjector /> should not render MathJax when hasLatex is false 1`] = `
<NewListPanels__ListItem
  theme={
    Object {
      "pale": "#999",
      "radius": Object {
        "general": "6px",
      },
    }
  }
>
  <HtmlInjector__HtmlBlock
    dangerouslySetInnerHTML={
      Object {
        "__html": "del testo",
      }
    }
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  />
</NewListPanels__ListItem>
`;

exports[`<HtmlInjector /> should not render MathJax when hasLatex is true 1`] = `
<NewListPanels__ListItem
  theme={
    Object {
      "pale": "#999",
      "radius": Object {
        "general": "6px",
      },
    }
  }
>
  <HtmlInjector__HtmlBlock
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  >
    <_default
      math="del testo"
    />
  </HtmlInjector__HtmlBlock>
</NewListPanels__ListItem>
`;

exports[`<HtmlInjector /> should render a ListItem 1`] = `
<NewListPanels__ListItem
  theme={
    Object {
      "pale": "#999",
      "radius": Object {
        "general": "6px",
      },
    }
  }
>
  <HtmlInjector__HtmlBlock
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  />
</NewListPanels__ListItem>
`;
